languages.adoc

```diff
@@ -18,88 +18,82 @@ SPDX-FileType: DOCUMENTATION
-| `factory` (or `builder`)
-| Factory methods to build the data model per business logic (i.e. to build CloudEvents, UUIDs, etc..)
+[.specitem,oft-sid="req~up-language-documentation~1",oft-needs="impl,utest"]
+--
+ * *MUST* contain a `README.adoc` that explains how to build and use the library. 
+--
 
-|===
+=== Test & CI
 
+[.specitem,oft-sid="req~up-language-test~1",oft-needs="impl,utest"]
+--
+ * *MUST* be able to run the test suite such that merge requests (commits) cannot be merged till tests are passed
+--
 
-=== Modules
-
-Modules are packages (folders) that reside under the `src` directory. <<content>> table below lists the required modules.
-
-.SDK Modules
-[#content,width="100%",cols="20%,20%,60%",options="header"]
-|===
-| Folder | Specification |Description
-
-| `uri`
-| link:basics/uri.adoc[UUri]
-| How things are addressed in uProtocol
-
-| `uuid`
-| link:basics/uuid.adoc[UUID]
-| Unique ID & timestamp specification 
-
-| `cloudevents`
-| link:up-l1/cloudevents.adoc[CloudEvents]
-| Specification of how uProtocol data models can be expressed using CloudEvents data model
-
-| `transport`
-| link:up-l1/README.adoc[uTransport]
-a| uP-L1 Transport Interface & data model.
-
-* *MUST* declare the interface & data model where as link:../upclient.adoc[uProtocol Client Library] implement the said interface.
-
-| `rpc`
-| link:up-l2/rpcclient.adoc[uP-L2 RPC Client]
-a|APIs used by clients to invoke methods to simplify implementation of the RPC design pattern. 
-
-|===
-
+[.specitem,oft-sid="req~up-language-ci-build~1",oft-needs="impl,utest"]
+--
+ * *MUST* have CI workflow that verifies pull requests compiles before they can be merged
+--
 
-== Test
+[.specitem,oft-sid="req~up-language-ci-test~1",oft-needs="impl,utest"]
+--
+ * *SHOULD* cover 100% of all code and specifications defined in <<Source>>
+--
 
- * *SHOULD* include a test suite under the folder `test`
- * *MUST* be able to run the test suite such that merge requests (commits) cannot be merged till tests are passed
-  * *SHOULD* cover 100% of all code and specifications defined in <<Source>>
+[.specitem,oft-sid="req~up-language-ci~3"]
+--
+ * *SHOULD* have linter for static code analysis as well as a tool that verifies APIs are properly documented
+--
 
 
-== Build
+=== Build
 
- * *MUST* be easily built in both a Windows and Linux development environment. Any build requirements (i.e. package dependencies) need to be properly documented in the README
-* *MUST* be able to be imported to any wellknown IDE (Eclipse, IntelliJ, VSCode, etc..) and include self contained build system to build the code and dependencies(ex. maven, gradle, clang, soong, etc...).
-* *MUST* fetch external dependency libraries as handle build dependencies, *MUST NOT* copy external dependencies
-* *MUST NOT* depend on external copyright or copyleft libraries/modules, only permissive licenses such as Apache 2.0, MIT, shall be accepted etc..
+[.specitem,oft-sid="req~up-language-build-sys~1",oft-needs="impl,utest"]
```

Comment:
Any idea how a unit test for this would/could look like? Maybe requiring an `impl` would be enough here? But we can change that later ...